octree/include/pcl/octree/octree_nodes.h

```diff
@@ -179,18 +178,12 @@ class OctreeLeafNode : public OctreeNode {
 template <typename ContainerT>
 class OctreeBranchNode : public OctreeNode {
 public:
-  /** \brief Empty constructor. */
-  OctreeBranchNode() : OctreeNode()
-  {
-    // reset pointer to child node vectors
-    child_node_array_ = {};
-  }
+  /** \brief Copy constructor. */
```

Comment:
This isn't a copy constructor?

Reply:
Looks as a copy constructor to me. Why do you think it is not?

Reply:
It doesn't take any old object of same type to copy from?

Reply:
Isn't that happening in line 189
child_node_array_[i] = source.child_node_array_[i]->deepCopy();
@mvieth : What's your opinion on that point?

Reply:
`OctreeBranchNode(const OctreeBranchNode& source)` is indeed a copy constructor. I think Lars was talking about `OctreeBranchNode()` a few lines earlier, which is not a copy constructor